### PR TITLE
increase minimal python version to 3.6

### PR DESCRIPTION
**Pull request by renefritze**

### CODE REVIEW

**Comment by renefritze** [comment]
closes #542 
closes #553 

I'd ask all @pymor/pymor-devs to look over the changes and see I missed any 3.5 -> 3.6 mentions.

File: .ci/appveyor.yml [hunk 1]
```diff
@@ -3,16 +3,16 @@ environment:
     # Pre-installed Python versions, which Appveyor may upgrade to
     # a later point release.
     # See: http://www.appveyor.com/docs/installed-software#python
-    - PYTHON_VERSION: "3.5" # currently 3.5.1
+    - PYTHON_VERSION: "3.6" # currently 3.5.1
```
Comment (pmli) [inline .ci/appveyor.yml]: @renefritze Do you want to update the `3.5.1` in the comment?

File: .ci/appveyor.yml [hunk 2]
```diff
@@ -3,16 +3,16 @@ environment:
...
       PYTHON_ARCH: "32"
-      CONDA: "C:\\Miniconda35"
+      CONDA: "C:\\Miniconda36"
       MARKER: "not grid"
       QT_API: "pyqt5"
       ALLOW_FAIL: true
 
-    - PYTHON_VERSION: "3.5" # currently 3.5.1
+    - PYTHON_VERSION: "3.6" # currently 3.5.1
```
Comment (pmli) [inline .ci/appveyor.yml]: See above.

File: .ci/appveyor.yml (hunk 1 above)
Comment (renefritze) [reply .ci/appveyor.yml]: Good catch! The appveyor builds will probably stay broken (/disabled) until I get to the conda packages for windows though.

**Comment by sdrave** [comment]
LGTM

**Comment by sdrave** [comment]
Didn't see that this also includes the f-string conversion. Have to look at this again ..

**Comment by sdrave** [comment]
Ok, so I think the automatic f-string conversion is way to conservative. [Here](https://github.com/pymor/pymor/commit/258098e1e8b9caa6c37e06fba3dc05f6d84b4712) you can find some manual conversions I did as a test. I think the result is much better ... -

**Comment by renefritze** [comment]
I'll look into hacking the converdion limit in pyupgrade then. Unless you want to continue manually changing more :wink:

**Comment by renefritze** [comment]
I've removed that restriction from pyupgrade and added another commit with more f-strings

File: src/pymor/core/cache.py [hunk 3]
```diff
@@ -271,7 +271,7 @@ def housekeeping(self):
                 ids_to_delete.append(id_)
                 files_to_delete.append(filename)
                 deleted += file_size
-            c.execute('DELETE FROM entries WHERE id in ({})'.format(','.join(map(str, ids_to_delete))))
+            c.execute(f'DELETE FROM entries WHERE id in ({""",""".join(map(str, ids_to_delete))})')
```
Comment (sdrave) [inline src/pymor/core/cache.py]: See above.

File: src/pymor/core/interfaces.py [hunk 4]
```diff
@@ -123,7 +123,7 @@ def __init__(cls, name, bases, namespace):
                 getattr(base, attribute).append(derived)
             else:
                 setattr(base, attribute, [derived])
-        cls._logger = logger.getLogger('{}.{}'.format(cls.__module__.replace('__main__', 'pymor'), name))
+        cls._logger = logger.getLogger(f'{cls.__module__.replace("__main__", "pymor")}.{name}')
```
Comment (sdrave) [inline src/pymor/core/interfaces.py]: better introduce a variable?

File: src/pymor/core/cache.py [hunk 5]
```diff
@@ -239,7 +239,7 @@ def clear(self):
         entries = c.fetchall()
         if entries:
             ids_to_delete, files_to_delete = zip(*entries)
-            c.execute('DELETE FROM entries WHERE id in ({})'.format(','.join(map(str, ids_to_delete))))
+            c.execute(f'DELETE FROM entries WHERE id in ({""",""".join(map(str, ids_to_delete))})')
```
[the remark this answers is not in the document]
Comment (renefritze) [reply src/pymor/core/cache.py]: No idea. You'd have to check the rendering in https://pypi.org/project/astor/

**Comment by sdrave** [comment]
There are still various f-string conversions ahead. Trying to finish this today ...

**Comment by sdrave** [comment]
I just finished the f-string conversions. If tests pass, we can merge IMHO.

**Comment by renefritze** [comment]
I don't quite understand how https://github.com/pymor/pymor/pull/553/files#diff-291c539e997b6c8ef74ac772372889adL409 could've ever worked? Was that just never evaluated?

**Pull request merged**